### PR TITLE
Fix mod Enabling / Disabling functionality

### LINKED ISSUE #427: Investigate mod Enable/Disable not working
Enabled state of mods doesn't seem to persist after closing application.
Needs further investigation.

### PR DESCRIPTION
**Pull request by Al12rs**
Fixes #427 

There were two separate issues: 
- The UI wasn't actually updating the sate of the mod as intended.
- The LoadoutSynchronizer wasn't excluding disabled mods during Apply step computations.

### CODE REVIEW

File: src/NexusMods.App.UI/RightContent/LoadoutGrid/Columns/ModEnabled/ModEnabledViewModel.cs [hunk 1]
```diff
@@ -59,7 +59,7 @@ public ModEnabledViewModel(LoadoutRegistry loadoutRegistry)
                 $"Setting {mod.Name} from {oldState} to {newState}",
                 mod =>
                 {
-                    if (mod?.Enabled == Enabled) return mod;
+                    if (mod?.Enabled == enabled) return mod;
```
Comment (erri120) [inline src/NexusMods.App.UI/RightContent/LoadoutGrid/Columns/ModEnabled/ModEnabledViewModel.cs:62]: Good catch!

**Review by erri120 — APPROVED** [review]

**Comment by Al12rs** [comment]
Fixed the tests and added a Disabled mod in the test setup to check that it doesn't break

File: tests/NexusMods.DataModel.Tests/LoadoutSynchronizerTests/IngestChangesTest.cs [hunk 2]
```diff
@@ -58,9 +58,10 @@ public async Task RemovedFilesAreRemoved()
             })
         });
 
-        loadout.Mods.Count.Should().Be(2);
+        loadout.Mods.Count.Should().Be(3);
 
         (from mod in loadout.Mods.Values
+            where mod.Enabled == true
             from file in mod.Files.Values
             select file).Count().Should().Be(2);
```
Comment (erri120) [inline tests/NexusMods.DataModel.Tests/LoadoutSynchronizerTests/IngestChangesTest.cs:66]: This should be converted into LINQ extensions: `loadout.Mods.Values.Where(...).Select(...).Count()...`

File: tests/NexusMods.DataModel.Tests/LoadoutSynchronizerTests/IngestChangesTest.cs [hunk 3]
```diff
@@ -78,9 +79,10 @@ public async Task RemovedFilesAreRemoved()
             }
         });
 
-        loadout.Mods.Count.Should().Be(2);
+        loadout.Mods.Count.Should().Be(3);
 
         (from mod in loadout.Mods.Values
+            where mod.Enabled == true
             from file in mod.Files.Values
             select file).Count().Should().Be(0);
```
Comment (erri120) [inline tests/NexusMods.DataModel.Tests/LoadoutSynchronizerTests/IngestChangesTest.cs:87]: Same here with LINQ extensions.

File: tests/NexusMods.DataModel.Tests/LoadoutSynchronizerTests/MakeIngestPlanTests.cs [hunk 4]
```diff
@@ -90,6 +90,7 @@ public async Task ChangedFilesAreUpdatedInTheLoadoutAndBackedUp()
         var absPath = loadout.Installation.Locations[GameFolderType.Game].Combine("0x00001.dat");
 
         var fileOne = (from mod in loadout.Mods
+            where mod.Value.Enabled == true
             from file in mod.Value.Files
             where file.Value is IFromArchive
             select (mod.Key, file.Key, file.Value)).First();
```
Comment (erri120) [inline tests/NexusMods.DataModel.Tests/LoadoutSynchronizerTests/MakeIngestPlanTests.cs:96]: Same here with LINQ extensions.

**Review by erri120 — CHANGES_REQUESTED** [review]

File: tests/NexusMods.DataModel.Tests/LoadoutSynchronizerTests/IngestChangesTest.cs (hunk 2 above)
Comment (Al12rs) [reply tests/NexusMods.DataModel.Tests/LoadoutSynchronizerTests/IngestChangesTest.cs:66]: Is there a specific advantage to using the Linq extension methods over the expression syntax?

**Pull request merged**